### ISSUE TITLE
False negatives for SC2086 - Double quote to prevent globbing and word splitting

### ISSUE THREAD

**Issue by ento** [hadolint/hadolint#984]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Running hadolint against this Dockerfile should report SC2086
```dockerfile
FROM debian:10.0
ARG FOO
RUN python foo.py --foo $FOO
```

### Actual behavior
SC2086 doesn't get reported

### Steps to reproduce the behavior

**Output of `hadolint --version` :**

Installed via nix-shell

```bash
$ hadolint --version
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile** (if relevant)

```Dockerfile
FROM debian:10.0
ARG FOO
RUN python foo.py --foo $FOO
```

**Additional environment details (OS, stack version, etc.)**

shellcheck's online playground reports SC2086 when you enter a similar script

```sh
#!/bin/sh
FOO=
python foo.py --foo $FOO
```

![image](https://github.com/hadolint/hadolint/assets/21108/c39584c5-8abd-4369-8da0-ab90845b3358)

Removing `ARG` makes hadolint report SC2086

```dockerfile
FROM debian:10.0 as debian
RUN python foo.py --foo $FOO
```

```sh
$ hadolint Dockerfile
Dockerfile:2 SC2086 info: Double quote to prevent globbing and word splitting.
```
